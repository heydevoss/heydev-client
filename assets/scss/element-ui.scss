// Import Custom Variables
@import "variables";

/* icon font path, required */
$--font-path: '~element-ui/lib/theme-chalk/fonts';

@import "~element-ui/packages/theme-chalk/src/index";