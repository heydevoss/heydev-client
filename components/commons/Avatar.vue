<template>
  <img :src="avatarUrl || require('~/assets/avatar/default.png')" />
</template>

<script>
export default {
  name: 'Avatar',
  props: {
    avatarUrl: String
  }
}
</script>
<style lang="scss" scoped>
img {
  border-radius: 100%;
  height: 300px;
  object-fit: cover;
  width: 300px;
}
</style>
