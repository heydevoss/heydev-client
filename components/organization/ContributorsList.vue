<template>
  <div>
    <ApolloQuery
      :query="require('@/apollo/queries/getContributors.gql')"
      :variables="{ quantity }"
    >
      <template v-slot="{ result: { loading, error, data } }">
        <div v-if="loading" v-loading="loading" class="loading apollo" />

        <div v-else-if="error" class="error apollo">An error occured</div>

        <!-- Result -->
        <div v-else-if="data" class="contributors">
          <div
            v-for="contributor in data.organization.contributors"
            :key="contributor.login"
            class="contributor"
          >
            <ContributorCard :contributor="contributor" />
          </div>
        </div>
        <div v-else v-loading="true" class="loading" />
      </template>
    </ApolloQuery>
    <el-button class="button--show-more" @click="showMoreContributors"
      >Show more</el-button
    >
  </div>
</template>
<script>
import ContributorCard from '@/components/organization/ContributorCard'
export default {
  name: 'ContributorsList',
  components: {
    ContributorCard
  },
  data() {
    return {
      quantity: 6
    }
  },
  methods: {
    showMoreContributors() {
      this.quantity += 3
    }
  }
}
</script>
<style lang="scss" scoped>
.contributors {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.contributor {
  margin: 3px 3px;
}

.loading {
  display: flex;
  justify-content: center;
}

.button--show-more {
  margin-top: 10px;
}
</style>
