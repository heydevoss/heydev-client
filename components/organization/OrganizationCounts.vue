<template>
  <div class="organization-counts">
    <div v-for="count in counts" :key="count.label" class="org-count">
      <div class="org-count--item">
        <span class="value">
          {{ count.value }}
        </span>
        <span class="label">
          {{ count.label }}
        </span>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'OrganizationCounts',
  props: {
    totalPullRequests: Number,
    totalCommits: Number,
    totalIssues: Number,
  },
  data() {
    return {
      counts: [
        { label: 'Issues', value: this.totalIssues },
        { label: 'Pull Requests', value: this.totalPullRequests },
        { label: 'Commits', value: this.totalCommits },
      ]
    }
  }
}
</script>

<style lang="scss">
.organization-counts {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}
.org-count {
  display: flex;
  flex-direction: column;
}
.org-count--item {
  display: flex;
  flex-direction: column;
}

.value {
  font-size: calc(2.0em + 1vw);
  color: #FFA600;
}

.label {
  margin-top: 4px;
  font-size: calc(0.2em + 1vw);
}

@media screen and (max-width: 820px) {
  .organization-counts {
    flex-direction: column;
    justify-content: space-between;
  }  
  .label {
    font-size: calc(1em + 1vw);
  }
}
</style>
