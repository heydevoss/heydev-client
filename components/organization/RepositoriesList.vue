<template>
  <div>
    <ApolloQuery
      :query="require('@/apollo/queries/getRepositories.gql')"
      :variables="{ quantity }"
    >
      <template v-slot="{ result: { loading, error, data } }">
        <div v-if="loading" v-loading="loading" class="loading apollo" />

        <div v-else-if="error" class="error apollo">An error occured</div>

        <!-- Result -->
        <div v-else-if="data" class="repositories">
          <div
            v-for="repository in data.organization.repositories"
            :key="repository.id"
            class="repository"
          >
            <RepositoryCard :repository="repository" />
          </div>
        </div>
        <div v-else v-loading="true" class="loading" />
      </template>
    </ApolloQuery>
    <el-button class="button--show-more" @click="showMoreRepositories"
      >Show more</el-button
    >
  </div>
</template>
<script>
import RepositoryCard from '@/components/organization/RepositoryCard'
export default {
  name: 'RepositoriesList',
  components: {
    RepositoryCard
  },
  data() {
    return {
      quantity: 6
    }
  },
  methods: {
    showMoreRepositories() {
      this.quantity += 3
    }
  }
}
</script>
<style lang="scss" scoped>
.repositories {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.repository {
  margin: 3px 3px;
  width: 300px;
}

.loading {
  display: flex;
  justify-content: center;
}
</style>
