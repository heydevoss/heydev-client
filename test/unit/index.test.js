import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import Index from '@/pages/index.vue'
import ContributorsList from '@/components/organization/ContributorsList'
import RepositoriesList from '@/components/organization/RepositoriesList'
import OrganizationCounts from '@/components/organization/OrganizationCounts'
import Avatar from '@/components/commons/Avatar'
import { Button } from 'element-ui'

Vue.use(Button)

describe('<Index/> Page', () => {
  const wrapper = shallowMount(Index, {
    data: () => ({
      organization: {
        name: 'Test',
        avatarUrl: '@/assets/avatar/default.png',
        url: 'www.github.com/testorg',
        login: 'testorg',
        websiteUrl: 'www.urltest.com',
        totalPullRequests: 50,
        totalCommits: 200,
        totalIssues: 1000
      }
    }),
    stubs: {
      Avatar,
      ContributorsList,
      RepositoriesList,
      OrganizationCounts,
      ApolloQuery: true,
      'el-button': Button
    }
  })

  it('should be a Vue instance and renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain organization avatar', () => {
    expect(wrapper.contains(Avatar)).toBe(true)
  })

  it('should contain organization name', () => {
    expect(
      wrapper.html().includes('<span class="org--name">Test</span>')
    ).toBeTruthy()
  })

  it('should contain organization counts data', () => {
    expect(wrapper.contains(OrganizationCounts)).toBeTruthy()
  })

  it('should contain organization repositories', () => {
    expect(wrapper.contains(RepositoriesList)).toBeTruthy()
  })

  it('should contain organization contributors', () => {
    expect(wrapper.contains(ContributorsList)).toBeTruthy()
  })

  it('should contain contributors section title', () => {
    expect(wrapper.html().includes('Contributors')).toBeTruthy()
  })

  it('should contain repositories section title', () => {
    expect(wrapper.html().includes('Repositories')).toBeTruthy()
  })
})
