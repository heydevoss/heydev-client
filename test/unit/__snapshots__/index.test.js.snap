// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index/> Page should be a Vue instance and renders correctly 1`] = `
<div
  class="container"
>
  <div
    class="info"
  >
    <img
      class="org--avatar"
      src="@/assets/avatar/default.png"
    />
     
    <div>
      <span
        class="org--name"
      >
        Test
      </span>
       
      <button
        class="org-info--github"
      />
       
      <div
        class="organization-counts org-info--counts"
      >
        <div
          class="org-count"
        >
          <div
            class="org-count--item"
          >
            <span
              class="value"
            >
              
        1000
      
            </span>
             
            <span
              class="label"
            >
              
        Issues
      
            </span>
          </div>
        </div>
        <div
          class="org-count"
        >
          <div
            class="org-count--item"
          >
            <span
              class="value"
            >
              
        50
      
            </span>
             
            <span
              class="label"
            >
              
        Pull Requests
      
            </span>
          </div>
        </div>
        <div
          class="org-count"
        >
          <div
            class="org-count--item"
          >
            <span
              class="value"
            >
              
        200
      
            </span>
             
            <span
              class="label"
            >
              
        Commits
      
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <section
    class="contributors"
  >
    <span
      class="section--title"
    >
      Contributors
    </span>
     
    <div>
      <apolloquery-stub
        query="[object Object]"
        variables="[object Object]"
      />
       
      <button
        class="el-button button--show-more el-button--default"
        type="button"
      >
        <!---->
        <!---->
        <span>
          Show more
        </span>
      </button>
    </div>
  </section>
   
  <section
    class="repositories"
  >
    <span
      class="section--title"
    >
      Repositories
    </span>
     
    <div>
      <apolloquery-stub
        query="[object Object]"
        variables="[object Object]"
      />
       
      <button
        class="el-button button--show-more el-button--default"
        type="button"
      >
        <!---->
        <!---->
        <span>
          Show more
        </span>
      </button>
    </div>
  </section>
</div>
`;
